ui/components/multichain/token-list-item/token-list-item.js

```diff
@@ -82,7 +88,41 @@ export const TokenListItem = ({
     title === CURRENCY_SYMBOLS.ETH && isOriginalTokenSymbol
       ? t('networkNameEthereum')
       : title;
-
+  const stakeableTitle = (
+    <Box
+      data-testid={`staking-entrypoint-${chainId}`}
+      display={Display.InlineFlex}
+      flexDirection={FlexDirection.Row}
+      alignItems={AlignItems.center}
+      gap={1}
+      onClick={(e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const url = getPortfolioUrl('stake', 'ext_stake_button', metaMetricsId);
```

Comment:
I don't think we should define url inside this onClick. wdyt?

Reply:
Hey @NidhiKJha, this was a harder decision, but I think the only time we use the variable is the very next line, so separating the declaration from the usage may reduce readability here. I'd move the declaration outside of scope if we used it in multiple scopes, but it seems a bit premature right now.